contentcuration/contentcuration/models.py

```diff
@@ -1777,6 +1777,14 @@ def mark_complete(self):  # noqa C901
         self.complete = not errors
         return errors
 
+    def make_content_id_unique(self):
+        """
+        If there exists a contentnode with the same content_id as the self,
+        then this updates content_id to a random uuid4.
+        """
+        if ContentNode.objects.exclude(pk=self.pk).filter(content_id=self.content_id).exists():
```

Comment:
I think this logic might be too simplistic.

The case I am thinking of here is when we make an update to a canonical content node. This logic is not running when we chef channels, so it doesn't apply to Khan Academy, for example, but if we had a manually created source channel then it would.

Imagine this example:

There is an exercise that has been created in a manually created Studio channel. This exercise is wildly popular, as it teaches a specific concept incredibly well, and gives rigorous practice on that concept.

After this has been copied thousands of times, the creator realizes there is a typo in one of the questions - they correct it. By this logic, the `content_id` of this exercise will now update. Kolibri users update their channel, seeing the update, and are surprised to discover that they no longer have any progress on the exercise they had previously mastered.

So, uniqueness is not quite the criterion needed here - we also need to consider whether it is a the 'originator' of this content_id - has this node been copied from any other node?

So, we should not update the `content_id` in the case that the `original_node_id` is `null` or equal to its own PK: https://github.com/learningequality/studio/blob/unstable/contentcuration/contentcuration/models.py#L1158

The other issue here is it ignores circumstances when two resources might legitimately share a content_id - two distinct nodes in the Khan Academy tree may have the same content_id, because they represent the same exercise, just copied into two locations in the topic tree (this is very common in Khan Academy).

Reply:
Conversation resolved on Slack and code pushed to address the concerns. :tada: 

Currently, writing unit tests for brittle paths coverage.